R/reporter-progress.R

```diff
@@ -62,18 +62,11 @@ ProgressReporter <- R6::R6Class("ProgressReporter",
 
     start_file = function(file) {
       self$file_name <- file
-
-      # Need to set here in case file doesn't contain any tests
       self$ctxt_issues <- Stack$new()
       self$ctxt_start_time <- proc.time()
-    },
 
-    start_test = function(context, test) {
-      if (is.null(context)) {
-        # Have to go via get_reporter() in order to trigger multi-reporter
-        name <- context_name(self$file_name)
-        get_reporter()$.start_context(name)
-      }
+      name <- context_name(self$file_name)
```

Comment:
I don't understand why this was previously only done in the first test, which reset any errors/failures captured earlier.

I guess there's a subtle change in behaviour if you had an empty test file? Previously it wouldn't have been shown; now it will be.

Reply:
Oooh, this is to ensure that if you do have a context, you don't get two context messages printed out

Reply:
Hmmm, not sure how to work around this. I only want to set the context in `start_file()` if it's not set elsewhere; maybe I should just grep the file for `context()` in the 2nd edition and earlier?

Reply:
Ok, I've just tweaked the display so the first context from the file will be over-written by the  context from `context()`. 

It's not quite right, because I'm getting an extra new line whenever the context takes more than `min_time` but I can't figure out where it's coming from.

Reply:
Yeah, I think the new code is cleaner.